Makefile

```diff
@@ -122,15 +126,18 @@ superclean: clean clean-luajit clean-ljsyscall
 # TODO: Move it to its own Makefile?
 #
 
-build-luajit $(luajit-inc) $(luajit-lib) $(luajit-exe):
+$(luajit-inc) $(luajit-lib) $(luajit-exe):
 	$(MAKE) -C $(luajit-dir) PREFIX=$(staging-dir)
 	$(MAKE) -C $(luajit-dir) PREFIX=$(staging-dir) install
+	# module search dir is not configurable, workaround with a symlink
+	ln -s $(luajit-jit) jit
+	rm $(staging-dir)/lib/*.so*
 
 clean-luajit:
-	$(MAKE) -C $(luajit-dir) PREFIX=$(staging-dir) uninstall
 	$(MAKE) -C $(luajit-dir) clean
+	rm -f jit
 
-.PHONY: build-luajit clean-luajit
```

Comment:
The phony build-luajit target was added just for manual testing just the luajit build.  We can drop it.